File: backend/app/schemas/common_area.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class CommonAreaBase(BaseModel):
    name: str
    description: Optional[str] = None
    capacity: Optional[int] = None
    opening_time: Optional[str] = None
    closing_time: Optional[str] = None
    is_active: bool = True

class CommonAreaCreate(CommonAreaBase):
    pass

class CommonAreaUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    capacity: Optional[int] = None
    opening_time: Optional[str] = None
    closing_time: Optional[str] = None
    is_active: Optional[bool] = None

class CommonAreaResponse(CommonAreaBase):
    id: str
    tenant_id: str

    class Config:
        from_attributes = True

File: frontend/admin/src/types/commonArea.ts
export interface CommonArea {
    id: string
    name: string
    description: string | null
    capacity: number | null
    opening_time: string | null
    closing_time: string | null
    is_active: boolean
    tenant_id: string
}

export interface CommonAreaCreate {
    name: string
    description?: string | null
    capacity?: number | null
    opening_time?: string | null
    closing_time?: string | null
    is_active?: boolean
}

export interface CommonAreaUpdate {
    name?: string
    description?: string | null
    capacity?: number | null
    opening_time?: string | null
    closing_time?: string | null
    is_active?: boolean
}
